smart_open/http.py

```diff
@@ -48,7 +48,12 @@ def __init__(self, url, mode='r', buffer_size=DEFAULT_BUFFER_SIZE,
         self.buffer_size = buffer_size
         self.mode = mode
 
-        self.response = requests.get(url, auth=auth, stream=True, headers=_HEADERS)
+        if headers:
```

Comment:
`if headers is None:` safer.

Like this, if a user explicitly passes `headers={}`, we'd ignore his choice and use `_HEADERS` anyway.